src/app/api/api_v1/endpoints/detections.py

```diff
@@ -49,6 +56,128 @@
 router = APIRouter()
 
 
+async def _attach_sequence_to_alert(
+    sequence_: Sequence,
+    camera: Camera,
+    cameras: CameraCRUD,
+    sequences: SequenceCRUD,
+    alerts: AlertCRUD,
+) -> None:
+    """Assign the given sequence to an alert based on cone/time overlap."""
+    org_cameras = await cameras.fetch_all(filters=("organization_id", camera.organization_id))
+    camera_by_id = {cam.id: cam for cam in org_cameras}
+
+    if sequence_.camera_id not in camera_by_id:
+        camera_by_id[sequence_.camera_id] = camera
+
+    # Fetch recent sequences for the organization based on recency of last_seen_at
+    recent_sequences = await sequences.fetch_all(
+        in_pair=("camera_id", list(camera_by_id.keys())),
+        inequality_pair=(
+            "last_seen_at",
+            ">",
+            datetime.utcnow() - timedelta(seconds=settings.SEQUENCE_RELAXATION_SECONDS),
+        ),
+    )
+
+    # Ensure the newly created sequence is present
+    if all(seq.id != sequence_.id for seq in recent_sequences):
+        recent_sequences.append(sequence_)
+
+    # Build DataFrame for overlap computation
+    records = []
+    for seq in recent_sequences:
+        cam = camera_by_id.get(seq.camera_id)
+        if cam is None or seq.cone_azimuth is None or seq.cone_angle is None:
+            continue
+        records.append({
+            "id": int(seq.id),
+            "lat": float(cam.lat),
+            "lon": float(cam.lon),
+            "cone_azimuth": float(seq.cone_azimuth),
+            "cone_angle": float(seq.cone_angle),
```

Comment:
(maybe to) Rename according to what is chosen as the attribute name in the database model.